### PR TITLE
feat(chat): iroh docs for chat 0

### PR DESCRIPTION
**Pull request by ludns**
- chore: migrate to big iroh package
- feat: docs

### CODE REVIEW

File: chat/src/logic.rs [hunk 1]
```diff
@@ -1,69 +1,157 @@
...
+        info!("setting up message subscription");
+        let doc = self.doc0.read().await;
+        let iroh = self.iroh.read().await;
+        if let (Some(doc), Some(iroh)) = (doc.as_ref(), iroh.as_ref()) {
+            let mut events = doc.subscribe().await?;
+            let message_watch_sender = self.message_watch.0.clone();
+            while let Some(Ok(event)) = events.next().await {
+                match event {
+                    LiveEvent::InsertRemote { .. } | LiveEvent::InsertLocal { .. } => {
+                        // This is kinda buggy because it seems stuff might not be reflected in the story immediately (also refetching everything here is stupid)
```
Comment (ludns) [inline chat/src/logic.rs:109]:
```suggestion
```

File: chat/src/logic.rs [hunk 2]
```diff
@@ -1,69 +1,156 @@
...
+    doc0: Arc<tokio::sync::RwLock<Option<Doc>>>,
+    messages: RwLock<Vec<Message>>,
     message_watch: (watch::Sender<()>, watch::Receiver<()>),
-    message_sender: Arc<Mutex<Option<GossipSender>>>,
     _auto_updater: AutoUpdater,
 }
 
-const TOPIC_ID: &str = "xyjjzwagm2c5cx56iltxtfcufdjejx2gxnyic7odeunwkndlygwa";
-const PEER_IDS: [&str; 1] = ["gawk3tiwjuvvuujgvpecpg4hcaheuzuzgnnghe4ljmz2r3zym4ya"];
+const DOP0: &str = "j2kblusfmti4mvc662jprvncnkes5qydcqj4bdl54ko6bdyfsx5a";
```
Comment (smallbraingames) [inline chat/src/logic.rs:31]: nit: "DOC"?

File: chat/src/logic.rs [hunk 3]
```diff
@@ -1,69 +1,156 @@
...
-        let messages = self.messages.clone();
-        let message_watch_sender = self.message_watch.0.clone();
-        if let Some(sender) = self.message_sender.lock().await.as_ref() {
+    async fn load_initial_messages(&self) -> anyhow::Result<()> {
+        info!("loading initial messages");
+        let iroh = self.iroh.read().await;
+        let doc = self.doc0.read().await;
+
+        if let (Some(iroh), Some(doc)) = (iroh.as_ref(), doc.as_ref()) {
+            let mut entries = doc.get_many(iroh::docs::store::Query::all()).await?;
```
Comment (smallbraingames) [inline chat/src/logic.rs:80]: this is rlly good

**Review by smallbraingames — APPROVED** [review]
preapproving just typo

**Pull request merged**